st2common/st2common/runners/paramiko_ssh.py

```diff
@@ -581,11 +584,97 @@ def _connect(self, host, socket=None):
 
         client = paramiko.SSHClient()
         client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        client.connect(**conninfo)
+
+        if cfg.CONF.ssh_runner.use_ssh_config:
+            conninfo_ssh_config = self._get_conninfo_from_ssh_config_for_host(host)
+            client = self._ssh_priority(client, conninfo, conninfo_ssh_config)
+
+        else:
+            extra = {'_conninfo': conninfo}
+            self.logger.debug('Connection info', extra=extra)
+            client.connect(**conninfo)
+
+        return client
+
+    @staticmethod
+    def _ssh_priority(client, conninfo, conninfo_ssh_config):
+        logger = logging.getLogger("ParamikoSSHClient")
+
+        # No Action params.
+        if conninfo['username'] == cfg.CONF.system_user.user\
```

Comment:
This method seems unnecessary complex to me.

I was envisioning something along the lines of having a method which retrieves parameters from ssh config file and then basically doing something like this:

```python
conn_info = {}

# 1. Retrieve parameters from ssh config
conn_info = self._get_parameters_from_ssh_config()

# 2. Add in arguments which are passed to the constructor. Those are action and runner parameters which have precedence over ones in the config
conn_info['username'] = username
...

client.connect(**conn_info)
```

This seems sufficient to me and we don't need any complex if statements, etc. If I'm missing / misunderstanding something, please let me know.